src/main/java/http_server/ClientSocket.java

```diff
@@ -0,0 +1,56 @@
+package http_server;
+
+import java.io.BufferedReader;
+import java.net.Socket;
+import java.io.IOException;
+import java.util.Arrays;
+
+public class ClientSocket implements SocketWrapper {
+    private final Socket clientSocket;
+    private BufferedReader inputStream;
+    private WriterWrapper outputStream;
+    char[] characterBuffer = new char[100000];
+
+    public ClientSocket(
+            Socket clientSocket,
+            BufferedReader inputStream,
+            WriterWrapper outputStream
+    ) {
+        this.clientSocket = clientSocket;
+        this.inputStream = inputStream;
+        this.outputStream = outputStream;
+    }
+
+    public String readData() {
+        try {
+            int bytes_read = inputStream.read(characterBuffer);
+            return new String(characterBuffer, 0, bytes_read);
+        } catch (IOException e) {
+            System.err.println(e.toString());
+            return e.toString();
+        }
+    }
```

Comment:
What happens here if you call `readData` more than once?

Reply:
I am not entirely sure, but because its trying to read the entire message at once now, I assume that calling it again means you've lost the previously sent message? Is that correct?

Reply:
I think so.  If you're not accessing the object / method more than once then this case doesn't ever really crop up, but on the other hand, if you _were_ in a spot where you needed to double-check the return value here then this function might surprise you.  Could probably solve the problem by just locally storing the return value after you've read it.  Not a big priority imo.